Makefile: add target for checking if pkg/lock or sync.Mutex is used

aalemayhu opened the pull request:
Makes the build fail if the lock package is not used.

Closes: #1702 (Add check to Makefile build to ensure that Cilium uses `pkg/lock` and not `sync.Mutex`)
Signed-off-by: Alexander Alemayhu <alexander@alemayhu.com>

aalemayhu committed:
Makefile: add target for checking if pkg/lock or sync.{,RW}Mutex is used

Makes the build fail if the lock package is not used. Adding the script
to contrib/ so it's easier to extend it later.

Closes: #1702 (Add check to Makefile build to ensure that Cilium uses `pkg/lock` and not `sync.Mutex`)
Signed-off-by: Alexander Alemayhu <alexander@alemayhu.com>

tgraf commented on Makefile:
```diff
@@ -229,5 +229,11 @@ install-manpages:
 	tests/00-check-cmdref.sh
 
+lock-check:
+	@if grep -r --exclude-dir={vendor,externalversions,lock} -i --include \*.go "sync.Mutex" .; \
```
You want to grep for both `sync.Mutex` and `sync.RWMutex`

aalemayhu committed:
pkg: replace sync.RWMutex with lock.RWMutex

Related: #1702 (Add check to Makefile build to ensure that Cilium uses pkg/lock and not sync.Mutex)
Signed-off-by: Alexander Alemayhu <alexander@alemayhu.com>

raybejjani reviewed (approved):
The CI failure seems to be a CLI <-> agent API parse thing, which I think we fixed on master.

joestringer reviewed (approved):
Looks like BPF codeowner was pulled in by prefilter.go. LGTM.
